### ISSUE TITLE
Can't use optional Float64 fields in models

### ISSUE THREAD

**Issue by perezperret** [luckyframework/lucky_record#295]
What I'm doing: Setting up a migration for a table with an optional float column (`Float64?`), and a model that uses this table, with an equivalent optional float field.

What I expect to happen: I get a model that can store a float if it needs to.

What actually happens: I get the following compiler error:
```
instantiating 'Float64::Lucky:Module#from_db!((PG::Numeric | Nil))'
in lib/lucky_record/src/lucky_record/type.cr:7: instantiating 'parse!((PG::Numeric | Nil))'

    parse!(value)
    ^~~~~~

in lib/lucky_record/src/lucky_record/type.cr:15: no overload matches 'Float64::Lucky.parse' with type (PG::Numeric | Nil)
Overloads are:
 - Float64::Lucky.parse(value : Float64)
 - Float64::Lucky.parse(value : String)
 - Float64::Lucky.parse(value : Int32)
 - Float64::Lucky.parse(value : Int64)
 - LuckyRecord::Type#parse(value : Nil)

    parse(value).as(SuccessfulCast).value
```

Things I already tried:
- My use case is a bit different but I distilled it down to a single column and a single field, both without any options.
- I tried leaving the column as optional and having the model's field be required and it works fine, do it looks like an issue with the model rather than the migration.
- I tried adding the same column as `Int32` and it works fine too.
- I'm looking into the source to see if I can get a PR up but it would be good to get some confirmation that this is a bug and perhaps some guidance on how to fix it if so.

In `spec/support/user.cr` I'm adding an item `'users.average_score'` and a column `average_score : Float64` and I get the same compiler error. Will this be enough of a test?

**Comment by paulcsmith** [comment]
Thank you so much for the thorough reporting. This is incredibly helpful. It does look like a bug. My guess is it needs a parse method here: https://github.com/luckyframework/lucky_record/blob/master/src/lucky_record/charms/float64_extensions.cr that accepts and parses a Pg::Numeric type. 

Feel free to comment here on ping me on Gitter if you need more info or want to discuss this more. Thanks so much for your help!

**Comment by perezperret** [comment]
I opened a PR but I still need some help, it's explained right there: https://github.com/luckyframework/lucky_record/pull/297

**Comment by paulcsmith** [comment]
Closed by #297
